server/controllers/user/signup.js

```diff
@@ -1,28 +1,30 @@
-const addUser = require('./../../database/query/addUser')
-const responsemessage = require('./../../helpers/responseMessage')
+const { addUser } = require('./../../database/query/user');
+const responsemessage = require('./../../helpers/responseMessage');
 const signup = (req, res) => {
-    console.log(req.body)
-    const data = req.body
-    addUser(data, (err, result) => {
-        console.log("err" + err)
-        console.log("result" + result)
-        if (err) {
-            return
-            res
-                .status(501)
-                .json(
-                    responsemessage.InternalErrorMessage(
-                        null,
-                        'internal error with the server'
-                    )
-                );
-        }
-        return
-        res
-            .status(200)
-            .json(responsemessage.successMessage(null, ' You are registered successfully'));
-    })
-
-
-}
-module.exports = signup;
\ No newline at end of file
+  const data = req.body;
```

Comment:
I see you have registrationValidation with Joi in another file, but where does that get applied? You should make sure that req.body is validated otherwise respond with a 400 error